mmv1/products/compute/InstanceSettings.yaml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'InstanceSettings'
+kind: 'compute#instanceSettings'
+base_url: projects/{{project}}/zones/{{zone}}/instanceSettings
+self_link: projects/{{project}}/zones/{{zone}}/instanceSettings
+description: |
+  Represents an Instance Settings resource. Instance settings are centralized configuration parameters that allow users to configure the default values for specific VM parameters that are normally set using GCE instance API methods.
+create_verb: :PATCH
+create_url: 'projects/{{project}}/zones/{{zone}}/instanceSettings?update_mask={{update_mask}}'
+update_verb: :PATCH
+update_url: 'projects/{{project}}/zones/{{zone}}/instanceSettings?update_mask={{update_mask}}'
+import_format: ['projects/{{project}}/zones/{{zone}}/instanceSettings']
+min_version: beta
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    kind: 'compute#operation'
+    path: 'name'
+    base_url: 'projects/{{project}}/zones/{{zone}}/operations/{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'targetLink'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'status'
+    complete: 'DONE'
+    allowed:
+      - 'PENDING'
+      - 'RUNNING'
+      - 'DONE'
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error/errors'
+    message: 'message'
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'instance_settings_basic'
+    primary_resource_id: 'gce_instance_settings'
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  custom_delete: templates/terraform/custom_delete/clear_instance_settings.go.erb
+  test_check_destroy: templates/terraform/custom_check_destroy/skip_delete_during_test.go.erb
+parameters:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'zone'
+    resource: 'Zone'
+    imports: 'name'
+    description: 'A reference to the zone where the machine resides.'
+    required: true
+    custom_expand: 'templates/terraform/custom_expand/resourceref_with_validation.go.erb'
+  - !ruby/object:Api::Type::String
+    name: 'update_mask'
+    required: true
+    immutable: true
+    description: |
+      The instance settings fields to be updated.
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::Fingerprint
+    name: 'fingerprint'
```

Comment:
is this field meant to be output only? The custom expand means the user supplied value will never be used, which will lead to potential permadiffs.

Reply:
We want to have the terraform behave similar to the [gcloud settings](https://cloud.google.com/compute/docs/metadata/setting-custom-metadata#procedure). So, instead of asking the user to supply fingerprint, it will be fetched by the Get command and will be used for the Update.

Reply:
I understand that -- which is why I ask if the field is is intended to be marked with output-only (as opposed to now where it is able to be marked by a user)

Reply:
I see. I can mark it as output : true. But, https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/type.rb#L533 Does this mean it is default true?

Reply:
ah yes my bad.